examples/stac-query.js

```diff
@@ -0,0 +1,433 @@
+
+      // Load collection
+      const collectionResult = await StacEndpoint.fromUrl(collectionUrl);
+      console.log(`   • Collection URL → Type: ${collectionResult.type}`);
+      console.log(`     Title: ${collectionResult.data.title}`);
+      console.log('');
+
+      // Load root/catalog
+      const catalogResult = await StacEndpoint.fromUrl(STAC_API_URL);
+      console.log(`   • Root/Catalog URL → Type: ${catalogResult.type}`);
+      console.log(
+        `     Title: ${catalogResult.data.title || catalogResult.data.id}`
+      );
+      console.log('');
+
+      // Load item
+      if (items.length > 0) {
+        const itemUrl = `${STAC_API_URL}/collections/${collectionId}/items/${items[0].id}`;
+        const itemResult = await StacEndpoint.fromUrl(itemUrl);
+        console.log(`   • Item URL → Type: ${itemResult.type}`);
+        console.log(`     ID: ${itemResult.data.id}`);
+        console.log(
+          `     Date: ${itemResult.data.properties.datetime || 'N/A'}`
+        );
+        console.log('');
+      }
+
+      console.log('   ✓ All resources loaded and auto-detected successfully\n');
+
+      // Demonstrate querying items with filtering from direct items URLs
+      console.log(`🔟  Querying items with filtering from direct URLs...`);
+      console.log(
+        '   These methods enable spatial/temporal filtering without root endpoint\n'
+      );
+
+      // Get the items URL from the collection
+      const itemsUrl = `${STAC_API_URL}/collections/${collectionId}/items`;
+
+      // Method 1: getItemsFromUrl - direct items URL query
+      console.log(`   a) Direct query with getItemsFromUrl:`);
+      console.log(`      Items URL: ${itemsUrl}`);
+      console.log(`      Filters: limit=3, bbox from first item`);
+
+      if (items.length > 0 && items[0].bbox) {
+        const directResponse = await StacEndpoint.getItemsFromUrl(itemsUrl, {
+          limit: 3,
+          bbox: filterBbox,
+        });
+        console.log(`      ✓ Found ${directResponse.features.length} item(s)`);
+        directResponse.features.forEach((item, idx) => {
+          console.log(
+            `         ${idx + 1}. ${item.id.substring(0, 40)}${
+              item.id.length > 40 ? '...' : ''
+            }`
+          );
+        });
+
+        // Show pagination links if available
+        const nextLink = directResponse.links?.find((l) => l.rel === 'next');
+        if (nextLink) {
+          console.log(`      → Next page available: ${nextLink.href}`);
+        }
+      } else {
+        console.log(`      ⊘ Skipped: No items with bbox available`);
+      }
+      console.log('');
+
+      // Method 2: getItemsFromCollection - two-step for flexibility
+      console.log(`   b) Two-step query with getItemsFromCollection:`);
+      console.log(`      First fetch collection, then query items`);
+
+      const collectionResult = await StacEndpoint.fromUrl(collectionUrl);
```

Comment:
I just checked out and wanted to run the example, but I got this error:
```
file:///home/lbruneton/dev/camptocamp/ogc-client/examples/stac-query.js:393
      const collectionResult = await StacEndpoint.fromUrl(collectionUrl);
            ^

SyntaxError: Identifier 'collectionResult' has already been declared
```
Seem to be already used from line 324.